src/ape/managers/query.py

```diff
@@ -76,12 +78,49 @@ def perform_block_transaction_query(
         return self.provider.get_transactions_by_block(query.block_id)
 
     @perform_query.register
-    def perform_contract_creation_query(self, query: ContractCreationQuery) -> Iterator[ReceiptAPI]:
-        yield from self.provider.get_contract_creation_receipts(
-            address=query.contract,
-            start_block=query.start_block,
-            stop_block=query.stop_block,
-        )
+    def perform_contract_creation_query(
+        self, query: ContractCreationQuery
+    ) -> Iterator[ContractCreation]:
+        def find_creation_block(lo, hi):
+            # perform a binary search to find where the contract code appeared
+            # takes log2(height), doesn't work with contracts that have been reinit.
+            while hi - lo > 1:
+                mid = (lo + hi) // 2
+                code = self.provider.get_code(query.contract, block_id=mid)
+                if not code:
+                    lo = mid
+                else:
+                    hi = mid
+
+            if self.provider.get_code(query.contract, block_id=hi):
+                return hi
+
+            return None
+
+        # if it doesn't have code at head, skip the search
+        if not self.provider.get_code(query.contract):
+            return None
+
+        block = find_creation_block(0, self.chain_manager.blocks.height)
+        traces = self.provider._make_request("trace_replayBlockTransactions", [block, ["trace"]])
```

Comment:
Before, nodes were **not** required to have `trace_replayBlockTransactions` RPC but now they are.
Is there a fallback we can do we here?

Reply:
@fubuloubu @banteg

Reply:
hmm, i think for geth we can try using `debug_traceBlockByNumber` with a custom tracer? i'll research this.

Reply:
working as before was incomplete as it couldn't find contracts deployed from factories. i guess this could be one of several fallbacks.

Reply:
Here is one course of action:

1. Have old way as a last-resort
2. Add a timeout to the last resort so it never feels too much like hanging
3. Log when using last resort about the factories (maybe as a DEBUG?)
4. Add a TODO + new issue for properly adding geth support that can get contracts from factories

Reply:
added a geth variant that implements essentially the same logic as the parity variant.

will add the last resort a bit later.